internal/utils/meta.go

```diff
@@ -31,3 +35,35 @@ func AddAnnotation(obj metav1.Object, key string, value string) bool {
 	}
 	return false
 }
+
+// IsOwnedBy returns true if the object has an owner ref for the provided owner
+func IsOwnedBy(obj, owner metav1.Object) bool {
+	ownerRefs := obj.GetOwnerReferences()
+	for i := range ownerRefs {
+		if owner.GetUID() == ownerRefs[i].UID {
+			return true
+		}
+	}
+	return false
+}
+
+// ToggleOwnerReference adds or remove an owner reference for the given owner based on the first argument.
+// The function return true if the owner reference list had changed and false it it didn't
+func ToggleOwnerReference(on bool, obj, owner metav1.Object, scheme *runtime.Scheme) (bool, error) {
+	ownerRefExists := IsOwnedBy(obj, owner)
+
+	if on {
+		if !ownerRefExists {
```

Comment:
any reason in not clubbing this condition?

Reply:
If not `on` we should execute `} else if ownerRefExists {` (line 62)
If `on` but  `!ownerRefExists` we should not execute `} else if ownerRefExists {` (line 62)

Reply:
Yes, `if on && !ownerRefExists` would result in the same because `on` acting as short-circuit, isn't it?

If we want to be elaborated it's fine, just when I saw, it occured to me that I might be missing something.

Reply:
@leelavg 
```
if on {
    if !ownerRefExists {
    }
} else ownerRefExists {
}
```
is not the same as 
```

if on && !ownerRefExists {
} else ownerRefExists {
}
```

Consider the case where on `on = true` and `ownerRefExists = true`. 
In the first snippet (my code) we will skip everything, with the second snippet (your proposal) we will run the code inside the else (because `ownerRefExists = true`)

Reply:
> we will skip everything
- may not skip everything, however I understood the explanation, resolve.